### ISSUE TITLE
Parameter Update Strategy for QAT with Backbone-only Quantization

### ISSUE THREAD

**Issue by piantou** [quic/aimet#4085]
Dear author,
I have a question when using AIMET for QAT training. For example, if I want to perform quantization on a model that consists of Backbone + Neck + Head, such as a YOLO-series model, and I only insert quantization nodes into the **Backbone**.
When I conduct QAT, should I update all the parameters of the **entire model**, or only fine-tune the parameters of the **Backbone**?
I look forward to your reply.

The reason for this question is that in this project, I have identified several different update strategies:
(1) In the code at https://github.com/quic/aimet/blob/0faa5c04ffab877009ec366fa15cdf2a9e3caee6/Examples/torch/v2/qat.py#L132:
```
optimizer = torch.optim.AdamW(
        # Only train quantization parameters,
        # keeping base model weights unchanged
        params={
            param
            for module in model.modules()
            for param in module.parameters()
            if isinstance(module, AffineQuantizerBase)
        },
        lr=0.001,
    )
```
It is clear that only the parameters of the quantization nodes are updated here.

(2) However, in the code at https://github.com/quic/aimet/blob/bd5016ca71bb54d8727a163428dbf1f2293ca8cb/Examples/torch/quantization/qat.ipynb, we can see the following code is used:
```
ImageNetDataPipeline.finetune(sim.model, epochs=1, learning_rate=5e-7, learning_rate_schedule=[5, 10], use_cuda=use_cuda)
```
In this case, all model parameters (both quantization nodes and non-quantization nodes) are fine-tuned.

**Comment by quic-kyunggeu** [comment]
Hi @piantou, that's a good theoretical question that we don't have fixed answers for.
Training the entire model is more aggressive than training only backbone, which has pros and cons.

* Pros: Might (but not guaranteed to) be able to yield **better accuracy**
* Cons: Training can become **unstable**, sometimes even failing to converge at all

Yet, these are just potential pros and cons at this point. If we really want to find out which one is better, all we can do is trying both options and comparing the results.

Overall, I'd like to recommend starting with a small set of learnable parameters, and gradually increase it if necessary.
In other words:
1. Start with quantizer-only training
(approach (1) in your question)
2. \+ backbone model parameters, **if** step 1 didn't yield good enough accuracy
3. \+ all model parameters, **if** step 2 didn't yield good enough accuracy
